bigbluebutton-html5/imports/ui/components/meeting-ended/service.ts

```diff
@@ -24,25 +24,72 @@ export const MeetingEndedTable = {
   ENDED_DUE_TO_SERVICE_INTERRUPTION: 'ENDED_DUE_TO_SERVICE_INTERRUPTION',
 };
 
+const findCommonDomain = (url1: string, url2: string): string => {
+  // Helper function to extract domain parts in reverse order
+  const getDomainParts = (url: string): string[] => {
+    try {
+      const { hostname } = new URL(url);
+      return hostname.split('.').reverse();
+    } catch (e) {
+      throw new Error(`Invalid URL format: ${url}`);
+    }
+  };
+
+  try {
+    const domain1Parts: string[] = getDomainParts(url1);
+    const domain2Parts: string[] = getDomainParts(url2);
+
+    // Find common parts starting from the end (TLD)
+    const commonParts: string[] = [];
+    const minLength: number = Math.min(domain1Parts.length, domain2Parts.length);
+
+    for (let i = 0; i < minLength; i += 1) {
+      if (domain1Parts[i] === domain2Parts[i]) {
+        commonParts.push(domain1Parts[i]);
+      } else {
+        break;
+      }
+    }
+
+    // Return the common parts in correct order
+    if (commonParts.length > 0) {
+      return commonParts.reverse().join('.');
+    }
+    return '';
+  } catch (error) {
+    return '';
+  }
+};
+
 export const openLearningDashboardUrl = (
   accessToken: string,
   mId: string,
   sToken:string,
   learningDashboardBase: string,
   lang: string,
 ) => {
-  if (accessToken && setLearningDashboardCookie(accessToken, mId)) {
+  if (accessToken && setLearningDashboardCookie(accessToken, mId, learningDashboardBase)) {
     window.open(`${learningDashboardBase}/?meeting=${mId}&lang=${lang}`, '_blank');
   } else {
     window.open(`${learningDashboardBase}/?meeting=${mId}&sessionToken=${sToken}&lang=${lang}`, '_blank');
   }
 };
 
-export const setLearningDashboardCookie = (accessToken: string, mId: string) => {
+export const setLearningDashboardCookie = (accessToken: string, mId: string, learningDashboardBase: string) => {
   if (accessToken !== null) {
     const lifetime = new Date();
     lifetime.setTime(lifetime.getTime() + (3600000)); // 1h (extends 7d when open Dashboard)
-    document.cookie = `ld-${mId}=${accessToken}; expires=${lifetime.toUTCString()}; path=/`;
+    let cookieString = `ld-${mId}=${accessToken}; expires=${lifetime.toUTCString()}; path=/`;
+
+    // In a cluster setup it will be necessary to specify the root domain
+    // because the Dashboard might be in a different subdomain
+    if (learningDashboardBase.startsWith('http')) {
```

Comment:
```suggestion
    if (learningDashboardBase && learningDashboardBase.startsWith('http')) {
```

to match https://github.com/bigbluebutton/bigbluebutton/pull/21522